pkg/resource/v1/chartconfig/resource.go

```diff
@@ -57,3 +76,35 @@ func New(config Config) (*Resource, error) {
 func (r *Resource) Name() string {
 	return Name
 }
+
+func prepareClusterConfig(baseClusterConfig cluster.Config, clusterGuestConfig v1alpha1.ClusterGuestConfig) (cluster.Config, error) {
+	var err error
+
+	// Use baseClusterConfig as a basis and supplement it with information from
+	// clusterGuestConfig.
+	clusterConfig := baseClusterConfig
+
+	clusterConfig.ClusterID = key.ClusterID(clusterGuestConfig)
+
+	clusterConfig.Domain.API, err = key.ServerDomain(clusterGuestConfig, certs.APICert)
```

Comment:
This looks like we can use `APIDomain` here directly because this here does exactly that. `ServerDomain` should then be private and public methods redirect to it key package internally.

Reply:
Yes I missed actually using APIDomain. Done now and I made serverDomain private. I'll do a follow up PR to use this in CertConfig.